### PR TITLE
[js/web] FP16 LayerNorm, InstanceNorm, SkipLayerNorm

### PR DESCRIPTION
**Pull request by dakenf**
### Description
This PR includes fixes for Norm operations to support FP16 and also some optimizations to use vec2/vec4 if possible

### CODE REVIEW

File: js/web/lib/wasm/jsep/webgpu/ops/layer-norm.ts [hunk 1]
```diff
@@ -55,40 +49,44 @@ const createLayerNormProgramInfo =
...
+          if (bias) {
+            variables.push(inputVariable('bias', bias.dataType, bias.dims, components));
+          }
+          variables.push(outputVariable('output', inputs[0].dataType, outputShape, components));
 
           const hasMeanDataOutput = outputCount > 1;
           const hasInvStdOutput = outputCount > 2;
-          let bindingIndex = 0;
+
+          if (hasMeanDataOutput) {
```
Comment (qjia7) [inline js/web/lib/wasm/jsep/webgpu/ops/layer-norm.ts:65]: Should we also use `DataType.float` for `Mean` output and `InvStdDev` output for `LayerNormalization` and cast the corresponding data to f32 (`CastToF32`) like you did in `SkipLayerNormalization`?

**Review by qjia7 — APPROVED** [review]
LGTM with one question.

File: js/web/lib/wasm/jsep/webgpu/ops/common.ts [hunk 2]
```diff
@@ -237,6 +237,52 @@ export const tensorTypeToWsglValueType = (type: DataType, components: 1|2|3|4 =
...
+  return 1;
+};
+
+/**
+ * A helper function that initializes variable as a scalar or vector. e.g. f32(0) or vec4f(0,0,0,0)
+ * @param dataType
+ * @param components
+ * @param value
+ */
+export const fillVector = (dataType = 'f32', components?: number, value = '0') => {
```
Comment (fs-eire) [inline js/web/lib/wasm/jsep/webgpu/ops/common.ts:261]: according to https://www.w3.org/TR/WGSL/#vec4-builtin, no need to use `vec4<f32>(x,x,x,x)` as it should be equivalent to `vec4<f32>(x)`

File: js/web/lib/wasm/jsep/webgpu/ops/layer-norm.ts (hunk 1 above)
Comment (dakenf) [reply js/web/lib/wasm/jsep/webgpu/ops/layer-norm.ts:65]: Yeah, I've missed that

**Comment by guschmue** [comment]
lint is nagging.

**Review by guschmue — DISMISSED** [review]

**Review by guschmue — APPROVED** [review]

**Pull request merged**